packages/lerna/src/utils.ts

```diff
@@ -14,23 +14,26 @@
  * limitations under the License.
  */
 
+import * as fs from "fs";
 import * as exec from "@actions/exec";
 
 export async function lernaList(onlyChanged?: boolean): Promise<Record<string, any>[]> {
-    let cmdOutput = await exec.getExecOutput("npx", ["lerna", "list", "--json", "--toposort"]);
-    const packageInfo = JSON.parse(cmdOutput.stdout);
-
+    const cmdArgs = ["lerna"];
     if (onlyChanged) {
-        cmdOutput = await exec.getExecOutput("npx", ["lerna", "changed", "--include-merged-tags"], { ignoreReturnCode: true });
-        const changedPackages = cmdOutput.stdout.split(/\r?\n/);
-        return packageInfo.filter((pkg: any) => changedPackages.includes(pkg.name));
+        cmdArgs.push("changed", "--include-merged-tags");
+    } else {
+        cmdArgs.push("list");
     }
```

Comment:
I love a cleaner implementation.
it seems like the `ignoreReturnCode` should handle the rest.
`exec.getExecOutput("npx", cmdArgs, { ignoreReturnCode: onlyChanged });` 
👍